### PR TITLE
Models refactoring

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

**Comment by AlekseySh** [comment]
Changelog:
* Added `from_pretrained` method to the interface of `IExtractor` (the formats of all the `pretrained_models` were unified, so they can be used by one parent method)
* Refactored code for `VitClipExtractor` (introduced constructors as functions, as in other classes)
* Reworked `ResnetExtractor`: added `resnet50_projector` as a separate constructor, all the pretrained classifier models added explicitly to `pretrained_models`, updated code & version for GradCam, removed unused code for `moco_v1`, added imagenet transforms for resnet classifiers and moco (table in `Zoo` was also updated with the results on new transforms) 
* Refactored tests for models creation, so there is no need for dedicated testing of `ExtractorWithMLP` anymore 
* Removed `strict_load` argument, it is always `True` since now

File: oml/models/resnet.py [hunk 1]
```diff
@@ -139,83 +150,62 @@ def calc_last_conv_channels(self) -> int:
...
 
         """
-        model_device = str(list(self.model.parameters())[0].device)
-        image_tensor = get_normalisation_albu()(image=image)["image"].to(model_device)
-        cam = GradCAM(model=self.model, target_layer=self.model.layer4[-1], use_cuda=model_device != "cpu")
-        gray_image = cam(image_tensor.unsqueeze(0), "gradcam", None)
+        if self.remove_fc:
+            raise ValueError("This method does not work if there is no FC layer in the model.")
+
+        need_transform = type(image) == TPILImage
```
Comment (AlekseySh) [inline oml/models/resnet.py:170]: need_convert

File: tests/test_oml/test_models/test_models_creation.py [hunk 2]
```diff
@@ -4,104 +4,68 @@
...
 )
-def test_creation(
-    constructor: IExtractor, args: Dict[str, Any], default_arch: str, download_large_checkpoints: bool
-) -> None:
+def test_extractor(constructor: IExtractor, args: Dict[str, Any]) -> None:
     # 1. Random weights
-    net = constructor(weights=None, arch=default_arch, **args)
-    net.forward(torch.randn(1, 3, 224, 224))
+    extractor = constructor(weights=None, **args).eval()
+    extractor.forward(torch.randn(1, 3, 224, 224))
```
Comment (AlekseySh) [inline tests/test_oml/test_models/test_models_creation.py:31]: check `.feat_dim`

File: tests/test_oml/test_models/test_models_creation.py [hunk 3]
```diff
@@ -4,104 +4,68 @@
...
 
     # 2. Load from file
-    fname = f"{default_arch}_random.pth"
-    torch.save({"state_dict": net.state_dict()}, fname)
-    net = constructor(weights=fname, arch=default_arch, **args)
-    net(torch.randn(1, 3, 224, 224))
+    fname = "weights_tmp.pth"
+    torch.save({"state_dict": extractor.state_dict()}, fname)
+    extractor = constructor(weights=fname, **args).eval()
+    extractor(torch.randn(1, 3, 224, 224))
```
Comment (AlekseySh) [inline tests/test_oml/test_models/test_models_creation.py:37]: add `all_close`

File: oml/models/projection.py [hunk 4]
```diff
@@ -13,47 +11,22 @@
...
-            mlp_features=[384],
-        )
-    }
-
     pretrained_models = {
         "vits16_224_mlp_384_inshop": {
             "url": f"{STORAGE_CKPTS}/inshop/vits16_224_mlp_384_inshop.ckpt",
             "hash": "35244966",
             "fname": "vits16_224_mlp_384_inshop.ckpt",
+            "init_args": {
```
Comment (DaloroAT) [inline oml/models/projection.py:25]: Those are bad changes. Class attributes are calculated when you enter the script with class definition even without creating instances.

You will create VitExtractor always when you try to import anything from this file.

Comment (DaloroAT) [reply oml/models/projection.py:25]: image

Comment (AlekseySh) [reply oml/models/projection.py]: I've reworked this, please check the new commit

**Pull request merged**